src/client/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -46,9 +49,19 @@ public bool CanInvokeBroker(CoreUIParent uiParent)
 
             uiParent.CallerViewController.InvokeOnMainThread(() =>
             {
-                result = UIApplication.SharedApplication.CanOpenUrl(new NSUrl(BrokerParameter.BrokerV2));
+                result = UIApplication.SharedApplication.CanOpenUrl(new NSUrl(BrokerParameter.BrokerV3));
+                if (result)
+                {
+                    BrokerRequestNonce = Guid.NewGuid().ToString();
```

Comment:
Needs refactoring. The method `CanInvokeBroker` suggest it is a `query` type of method (i.e. read some stuff, give me a result). However, you actually generate a nonce, so it creates state. 

I suggest you decouple the nonce creation from the CanInvoke method. If you are worried about performacen, then feel free to cache the values of `canInvokeV2Broker` and `canInvokeV3Broker` - but I would not be worried at this point.